src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -264,18 +265,29 @@ public void SetAll(bool value)
         ** Exceptions: ArgumentException if value == null or
         **             value.Length != this.Length.
         =========================================================================*/
-        public BitArray And(BitArray value)
+        public unsafe BitArray And(BitArray value)
         {
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
             if (Length != value.Length)
                 throw new ArgumentException(SR.Arg_ArrayLengthsDiffer);
 
-            for (int i = 0; i < m_array.Length; i++)
+            int i = 0;
+            if (Sse2.IsSupported && m_array.Length >= 4)
             {
-                m_array[i] &= value.m_array[i];
+                fixed (int* leftPtr = m_array)
+                fixed (int* rightPtr = value.m_array)
```

Comment:
nit: I think it is fine to stack multiply `fixed` statements, but I think we need braces around the `for` block